src/precice/impl/ParticipantImpl.cpp

```diff
@@ -436,13 +437,19 @@ void ParticipantImpl::handleDataAfterAdvance(bool reachedTimeWindowEnd, bool isT
     for (auto &context : _accessor->writeDataContexts()) {
       context.resetInitialGuesses();
     }
-    return;
+  } else {
+    // We are iterating
+    PRECICE_ASSERT(math::greater(timeSteppedTo, timeAfterAdvance), "We must have moved back in time!");
+
+    trimSendDataAfter(timeAfterAdvance);
   }
 
-  // We are iterating
-  PRECICE_ASSERT(math::greater(timeSteppedTo, timeAfterAdvance), "We must have moved back in time!");
+  if (reachedTimeWindowEnd) {
+    mapReadData(timeAfterAdvance);
+    performDataActions({action::Action::READ_MAPPING_POST});
+  }
```

Comment:
I am wondering and did not yet find a good answer: Does it make a difference that you now moved the "map" behind the `redetInitalGuesses`?
What's the motivation for changing the order here?

Reply:
First and foremost, the ordering wrt the reset was an oversight.

That said, doesn't it make sense to map the received data after convergence with a zero initial guess? After all, it belongs to the next time window.

Accidental correctness?

Reply:
Restored